### PR TITLE
[ADD] Product ISP

### LINKED ISSUE #35: [ADD] product_isp
This module allows you to define the international call services and use the phone rates.

# Usage

* Go to Sales > Catalog > Products
* Select or a create a product
* Select the type "Service"
* Check the box "International Call Service"

# Technical Specifications

## product.product

- [ ] Add a checkbox "International Call Service"
- [ ] Show the previous checkbox only if the type is service
- [ ] Add a smart button to show the phone rates and show it if International Call Service is true

### PR DESCRIPTION
**Pull request by osi-scampbell**
The following PR adds product_isp to the vertical-isp repo. 
Closes #35.

Depends on:

- [ ] PR #38

### CODE REVIEW

**Review by kvedder-amplex — COMMENTED** [review]
Did not get a chance to demo these in house yet, will have it done by days end 10/21/19

**Comment by max3903** [comment]
@kvedder-amplex FYI @osi-scampbell will rebase this branch so Runbot will be working here for testing.

File: product_isp/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+{
+    'name': 'Product ISP',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'summary': 'Assign ISPs to Products',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    "website": 'https://github.com/OCA/vertical-isp',
+    'depends': [
+        'stock',
```
Comment (dreispt) [inline product_isp/__manifest__.py]: Could it just depend on `product`?

File: product_isp/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    international_call_service = fields.Boolean("International Call Service",
```
Comment (dreispt) [inline product_isp/models/product_product.py:11]: I don't think we will need this field.

File: product_isp/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                                                default=False)
+    phone_rate_ids = fields.Many2many('phone.rate',
+                                      string="Phone Rates",
+                                      compute='_compute_phone_rate_ids')
+    phone_rate_count = fields.Integer(string="Phone Rates",
+                                      compute='_compute_phone_rate_ids')
```
Comment (dreispt) [inline product_isp/models/product_product.py:17]: These fields should go in `base_phone_rate`

File: product_isp/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                ref('base_phone_rate.action_phone_rate').read()[0]
+            rates = self.env['phone.rate'].search([])
+            if len(rates) == 1:
+                action['views'] = [
+                    (self.env.ref('base_phone_rate.phone_rate_form_view').id,
+                     'form')]
+                action['res_id'] = rates.id
+            else:
+                action['domain'] = [('id', 'in', rates.ids)]
+            return action
```
Comment (dreispt) [inline product_isp/models/product_product.py:39]: Should be in base_phone_rate

**Review by dreispt — CHANGES_REQUESTED** [review]
I don't think we need this module.
The features here should be included in #38.

**Comment by kvedder-amplex** [comment]
Im good with merging this into the other module.

**Comment by dreispt** [comment]
@kvedder-amplex A good plan today is better than a perfect plan tomorrow.
I can see ways to improve this, but we will go faster if we stick to Max's design for know.

I rebased this PR to (hopefully) get a green build.
I invite you to review and approve the PR.

**Review by dreispt — APPROVED** [review]

**Comment by kvedder-amplex** [comment]
I am good with this work. Feel free to proceed.

**Review by kvedder-amplex — APPROVED** [review]

**Pull request merged**